### ISSUE TITLE
"Export with name is already exported by stack" on prod environment

### ISSUE THREAD

**Issue by mdw2000** [aws-amplify/amplify-cli#2242]
I'm experiencing similar errors as in #1581 but with cli version `3.0.0` (after upgrading from `1.7.5`)

The strange thing is that `amplify push --yes` worked fine on `test` and `dev` environments after upgrading the cli, but failed on `prod` environment.

All environments was deployed using `1.7.5` before the upgrade to `3.0.0`

It errors with both Name and StreamArn exports: 

```
CREATE_FAILED      xxxxxxxx-1QKJVEI30DQPV-MediaTag-1UWVP2Q2EYMYK AWS::CloudFormation::Stack Wed Sep 11 2019 16:20:20 GMT+0200 (South Africa Standard Time) Export with name xxxxxxxxx:GetAtt:MediaTagDataSource:Name is already exported by stack xxxxxxx-1QKJVEI30DQPV-Tag-QUE8YVCJRBGV

```

```
CREATE_FAILED      xxxxxxx-1QKJVEI30DQPV-ServiceTag-D1SYJI48T3FE AWS::CloudFormation::Stack Wed Sep 11 2019 16:20:20 GMT+0200 (South Africa Standard Time) Export with name xxxxxxx:GetAtt:ServiceTagTable:StreamArn is already exported by stack xxxxxxxxx-1QKJVEI30DQPV-Tag-QUE8YVCJRBGV
```

The models mentioned in the errors above, but this happens with a few of the schema's many-to-many models 
 

```
type Tag @model(subscriptions: null) 
  @searchable
  @key(fields: ["company","id"]) {
    company: ID!
    id: String!
    ...
    ...
    services: [ServiceTag] @connection(name: "TagServiceTag")
    media: [MediaTag] @connection(name: "TagMediaTag")
}

type MediaTag @model(queries: null) @key(fields: ["company", "id"]) {
  company: ID!
  id: ID!
  media: Media @connection(name: "MediaMediaTag")
  tag: Tag @connection(name: "TagMediaTag")
}

type Media @model @searchable
@key(fields: ["company", "id"]) {
  company: String!
  id: ID!
  ...
  ...
  tags: [MediaTag] @connection(name: "MediaMediaTag")
}



type ServiceTag @model(queries: null) @key(fields: ["company", "id"]) {
  company: ID!
  id: ID!
  service: Service @connection(name: "ServiceServiceTag")
  tag: Tag @connection(name: "TagServiceTag")
}


type Service @model(subscriptions: null) @searchable @key(fields: ["company", "id"]){
  company: ID!
  id: ID!
  ...
  ...
  tags: [ServiceTag] @connection(name: "ServiceServiceTag")
}


```

**Comment by attilah** [comment]
@mdw2000 do I understand correctly that the issue is gone (for now), so we can close this issue?

**Comment by mdw2000** [comment]
@attilah - I managed to recreate the issue - it seems to be related to the `amplify env checkout` command when upgrading from 1.7.5 to 3.0.0.

- When the cli is upgraded before checking out an environment that was previously pushed with 1.7.5 it produces the errors when pushing.  
- When the cli is upgraded after checking out the env with 1.7.5 then the push with 3.0.0 works without errors


The following sequence produces the errors:
```
(cli currently at @1.7.5)
amplify env add testalpha
amplify push --yes

npm install -g @aws-amplify/cli@3.0.0
amplify env checkout test                    #<-- swith to another existing env to test the checkout back to testalpha
amplify env checkout testalpha
amplify push --yes
```

So checking out a `1.7.5` env with cli `3.0.0` and then pushing seems to trigger the issue.

Checking out the env with 1.7.5 and then upgrading the cli works without errors, e.g.

```
(cli @1.7.5)
amplify env add testalpha
amplify push --yes
amplify env checkout test
amplify env checkout testalpha

npm install -g @aws-amplify/cli@3.0.0
amplify push --yes
```

**Comment by attilah** [comment]
@mdw2000 I was trying to reproduce based on your steps, but on the last push with 3.x it does not push because no changes were detected. Aren't you missing something from there?

Although I'm pretty sure that a ```amplify api gql-compile``` before the push and after the env checkouts would fix it. Could you please try that and get back to me?

**Comment by kaustavghosh06** [comment]
@mdw2000 Are you still stuck on this issue?

**Comment by rpinna** [comment]
@kaustavghosh06: I believe this is still an issue.

I ran into the symptoms reported by @mdw2000, but not originally related to a cli update. That is, "Export with XXX already exported for stack ZZZ."  This was for both Name and StreamArn resources for all many-to-many model connections.

This was not originally associated with a cli update. I was originally deleting a development model and adding comments. When I found this report, I tried upgrading and still had the same error.

I did an amplify pull --restore to get back to a known good state. I made a whitespace change and did an api push. I received the same errors.

I downgraded to the original CLI, tried a whitespace change again and was able to do a push.

I'm using the downgraded CLI, so I'm not blocked in the short term, but I wanted to report the issue and offer to send you graph etc. via DM if it helps you troubleshoot.

Recapping:

originally on cli 1.6.11
change to graphql.schema (delete types and inputs (not model) and added comments
receive already exported errors discussed in this issue
upgrade cli to 1.12.0
amplify api push --yes
same errors
amplify env pull --restore
whitespace change
amplify apii push --yes
same errors
npm install -g @aws-amplify/cli@1.6.11
amplify env pull --restore
amplify env checkout env
whitespace changes
amplify api push --yes

Successful push
